DirectProgramming/DPC++/GraphTraversal/bitonic-sort/README.md

```diff
@@ -0,0 +1,120 @@
+- Step 1: for each new 2\*\*(n-2) bitonic sequences of size 4, (each new
+sequence consists of 2 consecutive previous sequences), it swaps the elements
+so that at the end of step 1, we have 2\*\*(n-2) bitonic sequences of size 4,
+and they follow an order: increasing, decreasing, increasing, ...,
+decreasing. Thus, they form 2\*\*(n-3) bitonic sequences of size 8.
+- Same logic applies until we reach the last step.
+- Step n: at this last step, we have one bitonic sequence of size 2\*\*n. The
+elements in the sequence are swapped until we have a sequence in increasing
+oder.
+
+The code will attempt first to execute on an available GPU and fallback to the system's CPU
+if a compatible GPU is not detected.
+
+## Key Implementation Details
+
+The basic DPC++ implementation explained in the code includes device selector, buffer, accessor, kernel, and command g
+roups. Unified Shared Memory (USM) is used for data management.
+
+## License  
+This code sample is licensed under MIT license  
+
+## Building the `bitonic-sort` Program for CPU and GPU
+
+### Running Samples In DevCloud
+If running a sample in the Intel DevCloud, remember that you must specify the compute node (CPU, GPU,
+FPGA) as well whether to run in batch or interactive mode. For more information see the Intel® oneAPI
+Base Toolkit Get Started Guide (https://devcloud.intel.com/oneapi/get-started/base-toolkit/)
+
+### On a Linux* System
+1. Build the program using the following `cmake` commands. 
+    ``` 
+    $ cd bitonic-sort
+    $ mkdir build
+    $ cd build
+    $ cmake ..
+    $ make
+    ```
+
+2. Run the program:
+    ```
+    make run
+    ```
+
+3. Clean the program using:
+    ```
+    make clean
+    ```
+
+### On a Windows* System
+    * Build the program using VS2017 or VS2019
+      Right click on the solution file and open using either VS2017 or VS2019 IDE.
+      Right click on the project in Solution explorer and select Rebuild.
+      From top menu select Debug -> Start without Debugging.
+
+    * Build the program using MSBuild
+      Open "x64 Native Tools Command Prompt for VS2017" or "x64 Native Tools Command Prompt for
+ VS2019"
+      Run - MSBuild bitonic-sort.sln /t:Rebuild /p:Configuration="Release"
+
+## Running the sample
+### Application Parameters
+	
+        Usage: bitonic-sort <exponent> <seed>
+
+where
+
+exponent is a positive number. The according length of the sequence is 2**exponent.
+
+seed is the seed used by the random generator to generate the randomness.
+
+The sample offloads the computation to GPU and then performs the computation in serial in the CPU.
+The results from the parallel and serial computation are compared. If the results are matched and
+the ascending order is verified, the application will display a “Success!” message.
+
+### Example of Output
+```
+$ ./bitonic-sort 21 47
+Array size: 2097152, seed: 47
+Device: Intel(R) Gen9 HD Graphics NEO
+Kernel time: 0.416827 sec
+CPU serial time: 0.60523 sec
+Success!
+```
```

Comment:
## Include Files
The include folder is located at "%ONEAPI_ROOT%\dev-utilities\latest\include" on your development system.